src/main/java/org/apidesign/WebSocket.java

```diff
@@ -0,0 +1,68 @@
+package org.apidesign;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apidesign.polyfill.Polyfill;
+import org.apidesign.polyfill.crypto.CryptoPolyfill;
+import org.apidesign.polyfill.timers.TimersPolyfill;
+import org.apidesign.polyfill.websocket.WebSocketPolyfill;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Source;
+import org.graalvm.polyglot.io.IOAccess;
+
+public class WebSocket {
+
+    private WebSocket() {
+    }
+
+    public static void initializePolyfill(Context ctx, ExecutorService executor) {
+        Polyfill[] components = new Polyfill[]{
+            new TimersPolyfill(executor),
```

Comment:
The _polyfill_ components have `initialize` method, but some of them (`TimersPolyfill`) need more to be initialized... hence I call this _over designed abstraction_.